packages/components/src/timeline/timeline-item.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import GridIcon from 'gridicons';
+
+const GRIDICON_SIZE = 16;
+
+const TimelineItem = ( props ) => {
+	const { item, className } = props;
+	const itemClassName = classnames( 'woocommerce-timeline-item', className );
+
+	const itemBody = item.body.map( ( line, bodyLineIndex ) => {
+		const bodyLineKey = item.datetime + '-' + bodyLineIndex;
+		return <span key={ bodyLineKey }>{ line }</span>;
+	} );
+	const itemTimeString = moment.unix( item.datetime ).format( 'h:mma' );
+
+	return (
+		<li className={ itemClassName }>
+			<div className={ 'woocommerce-timeline-item__top-border' }></div>
+			<div className={ 'woocommerce-timeline-item__title' }>
+				<p className={ 'woocommerce-timeline-item__headline' }>
+					<GridIcon icon={ item.gridicon } size={ GRIDICON_SIZE } />
+					<span>{ item.headline }</span>
+				</p>
+				<span className={ 'woocommerce-timeline-item__timestamp' }>
+					{ itemTimeString }
+				</span>
+			</div>
+			<p className={ 'woocommerce-timeline-item__body' }>{ itemBody }</p>
+		</li>
+	);
+};
+
+TimelineItem.propTypes = {
+	/**
+	 * Additional CSS classes.
+	 */
+	className: PropTypes.string,
+	/**
+	 * An array of list items.
+	 */
+	item: PropTypes.shape( {
```

Comment:
Hmmm, I think this might have to be a bit more complicated than providing a property for a class, or rather we need to reduce the scope and simplify our implementation.
If we just include a `class` prop that implies it will be applied to the whole item, i.e. both the icon and the associated text. So I think it'd be better to do one of two things:

1. Expand the `gridicon` prop to an object that has `icon` and `color` props. `icon` will contain the actual `<GridIcon />` element, whereas `color` contains the CSS color value.
2. The `gridicon` prop contains the `<GridIcon />` element, and we introduce a new prop, `iconColor` that will contain the CSS color value.

Personally I'm leaning towards option 2. What do you think?

Reply:
I think this has been properly implemented so far, what do you think @marcinbot ?